static/js/mld_store.js

```diff
@@ -0,0 +1,84 @@
+"use strict";
+
+class MldStore {
+    constructor() {
+        this._objects = new Map();
+    }
+
+    add(mld, key) {
+        if (key === undefined) {
+            key = hash_util.operators_to_hash(mld.filter.operators());
```

Comment:
I guess given this structure, we should have a plan for how to destroy these objects when they become invalid (basically the `change_narrow` code path) unless we're confident that we're updating data structures properly.